### DISCUSSION TITLE
Update from TeslaMate 1.27.4 -> 1.29.1

### DISCUSSION THREAD

**Discussion by abc123xxx** [teslamate-org/teslamate#3932] [Q&A]
How can i update my raspi 3 from TeslaMate 1.27.4 -> 1.29.1 without loosing my data?

**Comment by cwanja** [comment]
1. https://docs.teslamate.org/docs/maintenance/backup_restore (always)
2. https://docs.teslamate.org/docs/upgrading

**Comment by abc123xxx** [comment]
So first i make a backup, then update TeslaMate and then restore the data?

**Reply by cwanja** [reply]
The hope is you do not have to do a restore. But the instructions are there if you do need them.
